File: ui/src/main/java/com/saucedemo/pages/InventoryPage.java
package com.saucedemo.pages;

import com.microsoft.playwright.Locator;
import com.microsoft.playwright.Page;
import com.microsoft.playwright.options.AriaRole;

public class InventoryPage extends BasePage {
    private final Locator title     = locator(".title");
    private final Locator cartBadge = locator(".shopping_cart_badge");
    private final Locator cartIcon  = locator(".shopping_cart_link");
    private final Locator addBtns   = page.getByRole(AriaRole.BUTTON, new Page.GetByRoleOptions().setName("Add to cart"));

    public InventoryPage(Page page) {
        super(page);
    }

    public String getTitle() {
        return text(title);
    }

    public void addFirstItem() {
        click(addBtns.first());
    }

    public String cartCount() {
        return text(cartBadge);
    }

    public void goToCart() {
        click(cartIcon);
    }
}

File: ui.ts/pages/InventoryPage.ts
import { Locator, Page } from '@playwright/test';

export class InventoryPage {
  readonly page: Page;
  readonly addToCartButton: Locator;
  readonly cartBadge: Locator;
  readonly cartLink: Locator;
  readonly title: Locator;

  constructor(page: Page) {
    this.page = page;
    this.addToCartButton = page.locator('[data-test="add-to-cart-sauce-labs-backpack"]');
    this.cartBadge = page.locator('.shopping_cart_badge');
    this.cartLink = page.locator('.shopping_cart_link');
    this.title = page.locator('.title');
  }

  async addBackpackToCart() {
    await this.addToCartButton.click();
  }

  async goToCart() {
    await this.cartLink.click();
  }
}